.github/workflows/check-issue-links.yml

```diff
@@ -90,13 +88,14 @@ jobs:
           comment_text+="\nPlease manually unlink this PR from the following issues or reference the issues in a commit message: $extra_links"
         fi
 
-        echo $comment_text
-        
-        # Export the comment text as an environment variable.
-        echo "COMMENT_TEXT=$comment_text" >> $GITHUB_ENV
+        if [[ -n "$comment_text" ]]; then
+          echo $comment_text
 
+          # Export the comment text as an environment variable.
+          echo "COMMENT_TEXT=$comment_text" >> $GITHUB_ENV
+        fi
     - name: Add PR comment
-      if: always()
+      if: ${{ env.COMMENT_TEXT != '' }}
```

Comment:
Old comments should still be deleted, so either move the condition into the script or split the step.